### ISSUE TITLE
App crashes when you add a accounts own settings.

### ISSUE THREAD

**Issue by mrboomblaster** [laurencedawson/sync-for-lemmy#192]
**Describe the bug**  
When you add a account, the message will popup to run the setup to give the account its own settings. If you do that, the app will crash and keeps crashing when trying to open it. When you skip this, the app will work just fine.

**Steps to Reproduce**  
1. Go to Add account
2. Choose your instance and fill in your details
3. Click on "start setup" and go through it. I tried few different settings + leaving everything on default.
4. When finished the app can work normally for a moment but will crash eventually. If you close the app right after the setup and open it again, it will crash instantly.

**Expected behavior**  
Expecting to keep browsing lemmy with the new added instance.

**Smartphone (please complete the following information):**
 - Device: A54 samsung
 - Android: 13
 - Sync version: v23.08.01-20:30
 - Sync variant: beta + ultra

**Additional context**

**Comment by roschlau** [comment]
Happening to me as well, with an account from feddit.de, on a Pixel 7.

**Comment by mrboomblaster** [comment]
> Happening to me as well, with an account from feddit.de, on a Pixel 7.

Yeah, I think it's not the settings but the instance switching maybe? I use feddit.nl and when it wants to open it, the app crashes.

**Comment by kimboinatl** [comment]
I believe this is related to an issue I had. I tried changing my filter settings, and it crashes after that.

![Screenshot_20230802-132219](https://github.com/laurencedawson/sync-for-lemmy/assets/4192175/461763d0-500d-4be2-aa8d-977e64e543f2)

**Comment by jimmyhiggs337** [comment]
This occors on lemmy.world specifically for me. Presumably because of the issues they're having with comments, I would assume this is caused by the app trying to fetch something from the server without a try catch and expecting data that us null because of this.

Confirmed working imported settings on sh.itjust.works account

**Comment by ampz9** [comment]
Experiencing this as well after adding my account from Lemmy.world:

1. Open app after installing or clear data from settings
2. Go through the initial setup
3. Add account

It crashes after a few secs and it continues to crash after trying to launch again. Have to clear data (as cache it continues to crash)

Could it be the many subscriptions from various instances?

**Comment by mrboomblaster** [comment]
Installed te latest update but it still crashes. It seems to happen when I add another instance.

Anyway I can provide more information?

**Comment by ampz9** [comment]
Agreed, the new version did take longer to crash initially, but after it crashes, the only way to recover is to delete data and not login

**Comment by mrboomblaster** [comment]
Just installed the latest version, added feddit.nl and it crashes again. V23.08.02-2226 off

**Comment by mrboomblaster** [comment]
> Just installed the latest version, added feddit.nl and it crashes again. V23.08.02-2226 off

Yeah, it crashes as soon as I add feddit.nl. I had no problems when adding a different instance. So something is wrong with getting the information from that there?

**Comment by mrboomblaster** [comment]
I cleared my test account and made sure it was not subscribed to any communities.

I downloaded my profile of my main account to a json file with the help of the lasim program and used it add all my settings and communities to my test account.

Added the test account to sync, leaving everything to default, skipped notifications and custom settings for the profile.

Start using sync but it crashed.

So it must be a instance or community that's causing the problem. Now it's only finding out which one it is.
